File: src/main/java/de/rolf/games/cluedo/Verdacht.java
/**
 * 
 */
package de.rolf.games.cluedo;

import java.util.stream.Stream;
import lombok.Data;
import lombok.RequiredArgsConstructor;

/**
 * @author brf
 *
 */
@Data
@RequiredArgsConstructor
class Verdacht
{
   private final Spieler        erstelltVon;
   private Spieler              widerlegtVon;

   private final Karte.Taeter   taeter;
   private final Karte.Tatwaffe waffe;
   private final Karte.Tatort   tatort;

   boolean isWiderlegt()
   {
      return widerlegtVon != null;
   }

   Stream<Karte> getKarten() {
      return Stream.of(taeter, waffe, tatort);
   }

}

File: src/main/kotlin/de/rolf/games/cluedo/Verdacht.kt
package de.rolf.games.cluedo

import de.rolf.games.cluedo.Karte.Taeter
import de.rolf.games.cluedo.Karte.Tatort
import de.rolf.games.cluedo.Karte.Tatwaffe

data class Verdacht(val erstelltVon: Spieler, val taeter: Taeter, val waffe: Tatwaffe, val tatort: Tatort, var widerlegtVon: Spieler? = null) {

  val isWiderlegt: Boolean
    get() = widerlegtVon != null

  val karten: List<Karte>
    get() = listOf(taeter, waffe, tatort)
}